host/tests/dsHdmiIn_L3_Tests/dsHdmiIn_test2_SignalChangeCallback_Verify.py

```diff
@@ -0,0 +1,212 @@
+        """
+        Verifies whether the particular input selected or not.
+
+        Args:
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                     Defaults to other verification methods
+
+        Returns:
+            bool
+        """
+        if manual == True:
+            return self.testUserResponse.getUserYN(f'Is {port_type} Hdmi In device is ON? (Y/N):')
+        else :
+            #TODO: Add automation verification methods
+            return False
+
+    #TODO: Current version supports only manual verification.
+    def VerifyInput(self, manual=False):
+        """
+        Verifies whether the particular input selected or not.
+
+        Args:
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                     Defaults to other verification methods
+
+        Returns:
+            bool
+        """
+        if manual == True:
+            return self.testUserResponse.getUserYN("Is HdmiIn port selected? (Y/N):")
+        else :
+            #TODO: Add automation verification methods
+            return False
+
+    def find_SignalStatus(self, input_str: str, signal_status: str) -> bool:
+        if signal_status in input_str:
+            return True
+        return False
+
+    def testFunction(self):
+        """This Callback will check for Signal Change events on HdmiIn ports
+
+        Returns:
+            bool
+        """
+
+        # Download the assets listed in test setup configuration file
+        self.testDownloadAssets()
+
+        # Run Prerequisites listed in the test setup configuration file
+        self.testRunPrerequisites()
+
+        # Create the dsHdmiIn class
+        self.testdsHdmiIn = dsHdmiInClass(self.deviceProfile, self.hal_session)
+
+        self.log.testStart("test2_SignalChangeCallback_Verify", '1')
+
+        # Initialize the dsAudio module
+        self.testdsHdmiIn.initialise(self.testdsHdmiIn.getDeviceType())
+
+        audmix = 0      #default value false
+        videoplane = 0  #Always select primary plane.
+        topmost = 1     #Always should be true.
+   
+        # Loop through the supported HdmiIn ports
+        for port,index in self.testdsAudio.getSupportedPorts():
+            self.log.stepStart(f'Select {port} Port')
+            self.log.step(f'Select {port} Port')
+
+            # Check the HdmiIn device is active
+            result = self.CheckDeviceStatus(True,port)
+          
+            if result != False:
+                self.testdsHdmiIn.selectPort(port, index, audmix, videoplane, topmost)
+                result = self.VerifyInput(True)
+                self.log.stepResult(result, f'HdmiIn Select Verification {port} Port')
+                
+                result = self.hal_session.read_until("Received SignalChange status callback port:  , sigstatus:")
```

Comment:
move this handling inside the dsHdmiIn class

Reply:
done